src/s3-archive-sync.ts

```diff
@@ -0,0 +1,94 @@
+import * as path from 'path';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import { LambdaFunction } from '@aws-cdk/aws-events-targets';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import * as cr from '@aws-cdk/custom-resources';
+import { CommonEfsAssetsProps } from './common-props';
+
+export interface S3ArchiveFeederProps extends CommonEfsAssetsProps {
+  /**
+   * The S3 bucket containing the archive file.
+   */
+  readonly bucket: s3.IBucket;
+
+  /**
+   * The path of the zip file to extract in the S3 bucket.
+   */
+  readonly zipFilePath: string;
+
+  /**
+   * If this is set to true, then whenever a new object is uploaded to the specified path, an EFS sync will be triggered.
+   * Currently, this functionality depends on at least on CloudTrail Trail existing in your account that captures the S3
+   * event.
+   *
+   * (optional, default: true)
+   */
+  readonly shouldMonitorS3Path?: boolean;
```

Comment:
```suggestion
  readonly syncOnUpdate?: boolean;
```

How about `syncOnUpdate`, which is more straigit-forward?

Reply:
That makes sense, I went ahead and updated that.